Converting markdown files to html

FedG-code opened the pull request:
Added the marked js node.
Added converter.js which converts the markdown blogs into html.
Made new directories called html-blogs and markdown-blogs to separate the files properly.

FedG-code committed:
Converting markdown files to html

Added the marked js node.
Added converter.js which converts the markdown blogs into html.
Made new directories called html-blogs and markdown-blogs to separate the files properly.

robinglen commented on blogs/html-blogs/beyond-gitops_how-we-release-our-microservices-on-kubernetes-at-ynap.html:
```diff
@@ -0,0 +1,215 @@
+<p>﻿# Beyond GitOps: How we release our Microservices on Kubernetes at YNAP</p>
```
This should really be a `h1` not sure why its not the others seem be correct headers

FedG-code replied:
This one is confusing since the formatting in markdown is the same as the others.

robinglen commented on blogs/converter.js:
```diff
@@ -0,0 +1,33 @@
+const fs = require('fs');
+const marked = require('marked');
+
+var listOfDirectoryFiles = fs.readdirSync('markdown-blogs');
```
I think we should update the directories to the pattern
```
/blog
--html-blog.html
--/markdown
--/markdown/file.md
```
so markdown are inside a folder of blogs then the html files are at same level as that directory

FedG-code replied:
fixed.

robinglen commented on blogs/converter.js:
```diff
@@ -0,0 +1,33 @@
+const fs = require('fs');
+const marked = require('marked');
+
+var listOfDirectoryFiles = fs.readdirSync('markdown-blogs');
```
the difference with var, const and let

* var - is the old variable type, its dynamic
* const - its a constant cant be resigned
* let - is basically a replacement for var, can be resigned

`listOfDirectoryFiles` can be a const

FedG-code replied:
Yeah I think this was from when I was thinking of changing the original file.

robinglen commented on blogs/converter.js:
```diff
@@ -0,0 +1,33 @@
+
+console.log(listOfDirectoryFiles);
+
+var listOfMarkdownFiles = [];
```
All of this is good and works but to be more modern, you might want to look at more of the methods on an array.
`listOfDirectoryFiles` you might want to use a map to be more functional and cleaner:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map

FedG-code replied:
Since the markdown files are all in a separate folder I can skip this step now.

robinglen commented on blogs/converter.js:
```diff
@@ -0,0 +1,33 @@
+
+console.log(listOfMarkdownFiles);
+
+for(let i =0; i< listOfMarkdownFiles.length; i++){
```
Again here you could use the array type and a for each
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/forEach

FedG-code replied:
Done

robinglen commented on blogs/converter.js:
```diff
@@ -0,0 +1,33 @@
+   //pass the file through marked
+   var newHTMLFile = marked.marked(fileContent);
+   //get the new file name
+   const newFileName = currentFileName.slice(0,-2) + 'html';
```
this is a bit fragile, you could instead use the string method replace:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/replace

FedG-code replied:
All set. Definitely cleaner.

robinglen commented on blogs/converter.js:
```diff
@@ -0,0 +1,33 @@
+   //get the new file name
+   const newFileName = currentFileName.slice(0,-2) + 'html';
+   //write the html file with its new name
+   fs.writeFileSync('html-blogs/'+newFileName, newHTMLFile);
```
A better way of doing string concatenation is something called template literals.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals

FedG-code replied:
Again since the blogs are no longer stored in a separate folder this isn't necessary any more but I'll do it moving forward.

robinglen commented on blogs/converter.js:
```diff
@@ -0,0 +1,33 @@
+const fs = require('fs');
```
You can also use here:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment
so then for example you do
```
const { readdirSync } = require('fs');
```

FedG-code replied:
I'm not sure I understand this one, is it so you only pull reddirSync and not other functions from fs?

FedG-code replied:
don't I end up having to write this?

const {readdirSync} = require('fs');
const {readFileSync} = require('fs');
const {writeFileSync} = require('fs');

robinglen commented on blogs/converter.js:
```diff
@@ -0,0 +1,33 @@
+    //store the whole markdown file contents in currentFileBuffer
+   const currentFileBuffer = fs.readFileSync('markdown-blogs/' + currentFileName);
+    //Turn content to string
+    const fileContent = currentFileBuffer.toString();
```
instead of `toString` here, when you read the file you can read as `utf8`
```
fs.readFileSync('markdown-blogs/' + currentFileName, 'utf8`);
```
Then see template literal comment too

robinglen reviewed (commented):
Here is a review, the main thing is your code works... thats great... now lets take it further and look at some more advanced JS syntax

FedG-code committed:
Applied most fixes

Changed directory and file distribution.
Used cleaner string concatenation and replace methods.

Still missing: one of the files isn't converting the initial header correctly even though it looks fine in markdown and is the same as the others.  const {readdirSync} was used but likely improperly as it ended up using more code.
